### DISCUSSION TITLE
web3 Auth ?

### DISCUSSION THREAD

**Discussion by vexkiddy** [supabase/supabase#9499] [Feature Requests]
# Feature request

## Is your feature request related to a problem? Please describe.
We've been developing a web3 project and we're using supabase. We'd like to use your auth features so that they work with Web3 wallets like Metamask and others. Being able to Auth users based on these extensions would be great and a massive plus for new web3 products. 

So when a user 'connects' using their wallet and they sign a message, that this basically auths the user with a token / JWT in supabase.

**Comment by kiwicopple** [comment]
Hey @vexkiddy - I'll make this one a discussion.

Our friends at Wallet connect actually have a fork of our Auth server which includes web3 logins. You can read more about it here: https://medium.com/walletconnect/walletconnect-cloud-you-can-now-sign-in-with-ethereum-7563578686c2

We might merge this fork into our branch, but first we need to deliver MFA + SAML. Once we're confident that these are ready and secure, we'll investigate their fork.

**Comment by vexkiddy** [comment]
hiya @kiwicopple great! I should add a little more detail :D We're currently building a product on top of the LUKSO standards (more info on that can be found here: https://docs.lukso.tech/standards/universal-profile/introduction ) LUKSO uses something called Universal Profiles which are an upcoming standard to manage Identity for any EVM chain. 

Currently we're using their Alpha extension to sign messages and log-in to our product, so we're doing this kind of manually using our own table and JWT Tokens, however having an adapter for stuff like this going forward would be amazing :D I hope thats helpful in terms of giving some background to the issue :D
